menu/menu_displaylist.c

```diff
@@ -709,11 +709,30 @@ static int menu_displaylist_parse_core_info(
    if (core_info->firmware_count > 0)
    {
       core_info_ctx_firmware_t firmware_info;
-      bool update_missing_firmware   = false;
-      bool set_missing_firmware      = false;
+      uint8_t flags                   = content_get_flags();
+      bool update_missing_firmware    = false;
+      bool set_missing_firmware       = false;
+      bool systemfiles_in_content_dir = settings->bools.systemfiles_in_content_dir;
+      bool content_is_inited          = flags & CONTENT_ST_FLAG_IS_INITED;
+      char tmp_path[PATH_MAX_LENGTH];
 
       firmware_info.path             = core_info->path;
-      firmware_info.directory.system = settings->paths.directory_system;
+
+      /* If 'System Files are in Content Directory' is enabled and content is inited,
+       * adjust the path to check for firmware files */
+      if (systemfiles_in_content_dir && content_is_inited)
+      {
+         strlcpy(tmp_path, path_get(RARCH_PATH_CONTENT), sizeof(tmp_path));
+         path_basedir(tmp_path);
+
+         /* Removes trailing slash, doesn't really matter but it's more consistent with how
+          * the path is stored and displayed without 'System Files are in Content Directory' */
+         if (tmp_path[strlen(tmp_path) - 1] == PATH_DEFAULT_SLASH_C())
```

Comment:
Same thing here with strlen, this is wasteful.
(Solution 1) Either make it so that you only do strlen(tmp_path) once, having it do it two times like this is wasteful
(Solution 2) (Preferred but you will have to see how achievable this is) Not having to use the strlen at all and being able to figure out the size some other way beforehand. But this might not be achievable in which case you can go for solution 1.